src/EditorFeatures/Core/Classification/TotalClassificationTaggerProvider.cs

```diff
@@ -188,6 +196,24 @@ public override void AddTags(NormalizedSnapshotSpanCollection spans, SegmentedLi
 
         return;
 
+        bool PreferCurrentSemanticOverSyntacticTag()
+        {
+            if (currentSyntactic.Span == currentSemantic.Span)
+            {
+                // If they have the same classification type, then can def take one instead of both.
+                if (currentSyntactic.Tag.ClassificationType == currentSemantic.Tag.ClassificationType)
```

Comment:
>if (currentSyntactic.Tag.ClassificationType == currentSemantic.Tag.ClassificationType) [](http://example.com/codeflow?start=16&length=86)

If the base types are set correctly, couldn't this just check for derivation? eg, something like

if (currentSemantic.Tag.ClassificationType.IsOfType(currentSyntactic.Tag.ClassificationType.Classification))
    return true;

Reply:
Done.